internal/commands/chat-sast.go

```diff
@@ -38,75 +41,51 @@ func ChatSastSubCommand(chatWrapper wrappers.ChatWrapper, tenantWrapper wrappers
+func runChatSast(
+	chatWrapper wrappers.ChatWrapper, tenantWrapper wrappers.TenantConfigurationWrapper,
+) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		if !isAiGuidedRemediationEnabled(tenantWrapper) {
+		tenantConfigurationResponses, err := GetTenantConfigurationResponses(tenantWrapper)
+		if err != nil {
+			return outputError(cmd, uuid.Nil, err)
+		}
+		if !isAiGuidedRemediationEnabled(tenantConfigurationResponses) {
 			return outputError(cmd, uuid.Nil, errors.Errorf(AiGuidedRemediationDisabledError))
 		}
-		chatAPIKey, _ := cmd.Flags().GetString(params.ChatAPIKey)
 		chatConversationID, _ := cmd.Flags().GetString(params.ChatConversationID)
-		chatModel, _ := cmd.Flags().GetString(params.ChatModel)
 		scanResultsFile, _ := cmd.Flags().GetString(params.ChatSastScanResultsFile)
 		sourceDir, _ := cmd.Flags().GetString(params.ChatSastSourceDir)
 		sastResultID, _ := cmd.Flags().GetString(params.ChatSastResultID)
+		azureAiEnabled := isAzureAiGuidedRemediationEnabled(tenantConfigurationResponses)
+		checkmarxAiEnabled := isCheckmarxAiGuidedRemediationEnabled(tenantConfigurationResponses)
+		chatGptEnabled := isOpenAiGuidedRemediationEnabled(tenantConfigurationResponses)
 
-		statefulWrapper := wrapper.NewStatefulWrapper(connector.NewFileSystemConnector(""), chatAPIKey, chatModel, dropLen, 0)
-
-		newConversation := false
-		var userInput string
-		if chatConversationID == "" {
-			newConversation = true
-			chatConversationID = statefulWrapper.GenerateId().String()
-		} else {
-			userInput, _ = cmd.Flags().GetString(params.ChatUserInput)
-			if userInput == "" {
-				msg := fmt.Sprintf(UserInputRequiredErrorFormat, params.ChatUserInput, params.ChatConversationID)
-				logger.PrintIfVerbose(msg)
-				return outputError(cmd, uuid.Nil, errors.Errorf(msg))
-			}
-		}
+		statefulWrapper, customerToken := CreateStatefulWrapper(cmd, azureAiEnabled, checkmarxAiEnabled, tenantConfigurationResponses)
 
-		id, err := uuid.Parse(chatConversationID)
-		if err != nil {
-			logger.PrintIfVerbose(err.Error())
-			return outputError(cmd, id, errors.Errorf(ConversationIDErrorFormat, chatConversationID))
-		}
+		tenantID := getTenantID(customerToken)
 
-		var newMessages []message.Message
-		if newConversation {
-			systemPrompt, userPrompt, e := sastchat.BuildPrompt(scanResultsFile, sastResultID, sourceDir)
-			if e != nil {
-				logger.PrintIfVerbose(e.Error())
-				return outputError(cmd, id, e)
-			}
-			newMessages = append(newMessages, message.Message{
-				Role:    role.System,
-				Content: systemPrompt,
-			}, message.Message{
-				Role:    role.User,
-				Content: userPrompt,
-			})
-		} else {
-			newMessages = append(newMessages, message.Message{
-				Role:    role.User,
-				Content: userInput,
-			})
+		newConversation, userInput, id, err4, done3 := getSastConversationDetails(cmd, chatConversationID, statefulWrapper)
+		if done3 {
```

Comment:
please change to more indicative name: sastConversationErr, sastConversationSuccess and etc